src/coreclr/src/jit/valuenum.cpp

```diff
@@ -6558,18 +6558,17 @@ void Compiler::fgValueNumberBlockAssignment(GenTree* tree)
                     {
                         if (srcAddrFuncApp.m_func == VNF_PtrToStatic)
                         {
-                            var_types indType    = lclVarTree->TypeGet();
-                            ValueNum  fieldSeqVN = srcAddrFuncApp.m_args[0];
-
+                            var_types     indType            = lclVarTree->TypeGet();
+                            ValueNum      fieldSeqVN         = srcAddrFuncApp.m_args[0];
+                            FieldSeqNode* fldSeqForStaticVar = vnStore->FieldSeqVNToFieldSeq(fieldSeqVN);
+#ifdef DEBUG
                             FieldSeqNode* zeroOffsetFldSeq = nullptr;
                             if (GetZeroOffsetFieldMap()->Lookup(srcAddr, &zeroOffsetFldSeq))
                             {
-                                fieldSeqVN =
-                                    vnStore->FieldSeqVNAppend(fieldSeqVN, vnStore->VNForFieldSeq(zeroOffsetFldSeq));
+                                // Check that the zero offset field seq was attached for `srcAddr`.
```

Comment:
There was an x86 assert failure, that Brian helped me to debug and fix, the PR was updated.
I have also run SPMI tests over pri1 and internal collections and they have not shown any other asserts.

The previous code was added before TFS so I don't know the context, but I suspect it was dead code because we were checking this `GetZeroOffsetFieldMap` for `srcAddr` twice and it was either both `true` and then the assert should happen, of both `false` and we did not pass the check.